workflows/export/asset_export-vx.go

```diff
@@ -88,38 +104,161 @@ func AssetExportVX(ctx workflow.Context, params AssetExportParams) (*AssetExport
 
 	var mergeResult MergeExportDataResult
 	err = workflow.ExecuteChildWorkflow(ctx, MergeExportData, MergeExportDataParams{
-		ExportData: data,
-		TempPath:   tempFolder,
-		OutputPath: outputFolder,
+		ExportData:   data,
+		TempDir:      tempDir,
+		SubtitlesDir: outputDir,
 	}).Get(ctx, &mergeResult)
 	if err != nil {
 		return nil, err
 	}
 
+	// Destination branching:  VOD, playout, bmm, etc.
+	var results []*AssetExportResult
+	for _, dest := range params.Destinations {
+		destination := AssetExportDestinations.Parse(dest)
+		if destination == nil {
+			return nil, fmt.Errorf("invalid destination: %s", dest)
+		}
+		switch *destination {
+		case AssetExportDestinationVOD:
+			var result *AssetExportResult
+			ctx = workflow.WithChildOptions(ctx, workflows.GetDefaultWorkflowOptions())
+			err := workflow.ExecuteChildWorkflow(ctx, ExportToVOD, ExportToVODParams{
+				ParentParams: params,
+				ExportData:   *data,
+				MergeResult:  mergeResult,
+				TempFolder:   tempDir,
+				OutputFolder: outputDir,
+			}).Get(ctx, &result)
+			if err != nil {
+				return nil, err
+			}
+			results = append(results, result)
+		case AssetExportDestinationPlayout:
+			var result *AssetExportResult
+			ctx = workflow.WithChildOptions(ctx, workflows.GetDefaultWorkflowOptions())
+			err := workflow.ExecuteChildWorkflow(ctx, ExportToPlayout, ExportToPlayoutParams{
+				ParentParams: params,
+				ExportData:   *data,
+				MergeResult:  mergeResult,
+				TempDir:      tempDir,
+				OutputDir:    outputDir,
+			}).Get(ctx, &result)
+			if err != nil {
+				return nil, err
+			}
+			results = append(results, result)
+		}
```

Comment:
Could there just be 2 params set here: `workflow` and `workflowParams`? Then call the `WithChildOptions` and `ExecuteChildWorkflow` in one place.
The thing accepts interfaces, so the variables could be defined in those terms, no?

Reply:
Or even just one? The params look the same atm.